kb/2025-09-30/setup_common_cpu_model_for_vm_live_migration.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Setting up a common CPU model for virtual machine migration
+description: 'How to set up a common CPU model for virtual machine migration'
+slug: setup_common_cpu_model_for_vm_live_migration
+authors:
+  - name: Jack Yu
+    title: Senior Software Engineer
+    url: https://github.com/Yu-Jack
+    image_url: https://github.com/Yu-Jack.png
+tags: [harvester, virtual machine, VM, live migration, policy, strategy, configuration]
+hide_table_of_contents: false
+---
+
+## Problem Description
+
+Although Harvester allows you to perform live migration, it sometimes fails due to mismatched CPU features and models. Let's examine an example.
+
+When a virtual machine is migrated to another node, the following key-value pairs are added under `spec.nodeSelector` in the POD spec.
+
+```yaml
+spec:
+  nodeSelector:
+    cpu-model-migration.node.kubevirt.io/SierraForest: "true"
+    cpu-feature.node.kubevirt.io/fpu: "true"
+    cpu-feature.node.kubevirt.io/vme: "true"
+```
+
+After the first migration, if you migrate the virtual machine again, it will still use the above `nodeSelector` in the POD spec. Migration may fail if another node doesn't have the corresponding features or model. For example:
+
+```yaml
+# Node A
+labels:
+  cpu-model-migration.node.kubevirt.io/SierraForest: "true"
+  cpu-feature.node.kubevirt.io/fpu: "true"
+  cpu-feature.node.kubevirt.io/vme: "true"
+
+# Node B
+labels:
+  cpu-model-migration.node.kubevirt.io/SierraForest: "true"
+  cpu-feature.node.kubevirt.io/vme: "true"
+```
+
+This virtual machine will fail to migrate to node B due to the missing `fpu` feature. However, if the virtual machine doesn't actually require this feature, this failure can be frustrating. Therefore, setting up a common CPU model can resolve this issue.
+
+## How to Set Up a Common CPU Model
+
+In Harvester, we use the default CPU model, which is `host-model`. Using the default CPU model results in the above consequences. However, we can actually set up a customized CPU model in the virtual machine spec to make it migrate successfully.
```

Comment:
We can remove this since I moved this up to the intro. This gives the reader an idea of the main cause of the issue i.e. the default `host-model` configuration.